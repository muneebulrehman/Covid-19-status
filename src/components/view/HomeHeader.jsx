import React from 'react';
import { useSelector } from 'react-redux';
import style from '../styling/HomeHeader.module.css';

const HomeHeader = () => {
  const date = new Date();
  const today = `${date.getFullYear()}-${String(date.getMonth() + 1).padStart(2, '0')}-${String(
    date.getDate()
  ).padStart(2, '0')}`;
  const data = useSelector(({ covidReducer }) => covidReducer.allData);
  if (data) {
    return (
      <div className={style.homeHeader}>
        <div className={style.imgContainer}>
          <h3>{today}</h3>
          <p>STATS</p>
        </div>
        <div className={style.headerInfo}>
          <h3 className={style.heading}>TOTAL STATS</h3>
          <div className={style.headerData}>
            <p>
              Total Confirmed <br /> {data.today_confirmed}
            </p>
            <p>
              Total Deaths <br /> {data.today_deaths}
            </p>
            <p>
              Total cases for Today <br /> {data.today_new_confirmed}
            </p>
            <p>
              Total deaths for Today <br /> {data.today_new_deaths}
            </p>
          </div>
        </div>
      </div>
    );
  }
  return <h2>No data found</h2>;
};

export default HomeHeader;
