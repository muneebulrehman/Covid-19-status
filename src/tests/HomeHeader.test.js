import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import HomeHeader from '../components/view/HomeHeader';
import store from '../redux/configureStore';

describe('Testing the Home header', () => {
  it('Home header rendered correctly', () => {
    const home = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <HomeHeader />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(home).toMatchSnapshot();
  });
});
