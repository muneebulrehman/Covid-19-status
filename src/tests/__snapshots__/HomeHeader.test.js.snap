// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home header Home header rendered correctly 1`] = `
<div
  className="homeHeader"
>
  <div
    className="imgContainer"
  >
    <h3>
      2022-04-12
    </h3>
    <p>
      STATS
    </p>
  </div>
  <div
    className="headerInfo"
  >
    <h3
      className="heading"
    >
      TOTAL STATS
    </h3>
    <div
      className="headerData"
    >
      <p>
        Total Confirmed 
        <br />
         
      </p>
      <p>
        Total Deaths 
        <br />
         
      </p>
      <p>
        Total cases for Today 
        <br />
         
      </p>
      <p>
        Total deaths for Today 
        <br />
         
      </p>
    </div>
  </div>
</div>
`;
